### PR TITLE
HR2 module 

### PR DESCRIPTION
**Pull request by pranavelric**
Employee models and basic UI created for HR2 module

### CODE REVIEW

**Comment by pranavelric** [comment]
added database model in application/hr2/models.py

File: FusionIIIT/applications/hr2/models.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    )
+
+
+
+
+
+# table for employee details
+# Employee model
+def employee(models.Model):
```
Comment (adi0311) [inline FusionIIIT/applications/hr2/models.py]: This should be a class. Class name should be in PascalCase.

File: FusionIIIT/applications/hr2/models.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    id = models.OneToOneField(ExtraInfo, on_delete=models.CASCADE, primary_key=True)
```
Comment (adi0311) [inline FusionIIIT/applications/hr2/models.py]: Change the name of this field.

File: FusionIIIT/applications/hr2/models.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    first_name = models.CharField(max_length=40, default='')
+    middle_name = models.CharField(max_length=40, default='')
+    last_name = models.CharField(max_length=40, default='')
+    father_name = models.CharField(max_length=40, default='')
+    mother_name = models.CharField(max_length=40, default='')
+    gender = models.CharField(max_length=1, choices=Constants.GENDER_CHOICES)
```
Comment (adi0311) [inline FusionIIIT/applications/hr2/models.py]: Redundant field, already present in ExtraInfo.

File: FusionIIIT/applications/hr2/models.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        max_length=50, null=False, choices=Constants.CATEGORY)
+    cast  = models.CharField(max_length=40, default='')
+    home_state =   models.CharField(max_length=40, default='')
+    home_district =  models.CharField(max_length=40, default='')
+    identification_mark =  models.CharField(max_length=40, default='')
+    height  =  models.IntegerField(default=0)
+    dob =  models.DateTimeField(max_length=6, null=True)
+    date_of_joining =  models.DateTimeField(max_length=6, null=True)
+    designation =  models.CharField(max_length=40, default='')
+    blood_group =  models.CharField(max_length=40, default='')
```
Comment (adi0311) [inline FusionIIIT/applications/hr2/models.py]: Provide Choices.

**Review by adi0311 — CHANGES_REQUESTED** [review]

File: FusionIIIT/applications/hr2/models.py (hunk 4 above)
Comment (adi0311) [reply FusionIIIT/applications/hr2/models.py]: Change all the models from functions to class(change their naming convention from snake_case to PascalCase), remove all the redundant fields and provide choices where possible.

File: FusionIIIT/applications/hr2/models.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+
+
+
+# table for employee details
+# Employee model
+class Employee(models.Model):
+
+    extra_info = models.OneToOneField(ExtraInfo, on_delete=models.CASCADE)
+    first_name = models.CharField(max_length=40, default='')
```
Comment (adi0311) [inline FusionIIIT/applications/hr2/models.py]: Already present in default User model of Django.

File: FusionIIIT/applications/hr2/models.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    middle_name = models.CharField(max_length=40, default='')
+    last_name = models.CharField(max_length=40, default='')
```
Comment (adi0311) [inline FusionIIIT/applications/hr2/models.py]: This also.

File: FusionIIIT/Fusion/urls.py [hunk 7]
```diff
@@ -47,4 +47,6 @@
     url(r'^estate/', include('applications.estate_module.urls')),
     url(r'^accounts/', include('allauth.urls')),
     url(r'^__debug__/', include(debug_toolbar.urls)),
+    # new
```
Comment (adi0311) [inline FusionIIIT/Fusion/urls.py]: Remove this line.

File: FusionIIIT/applications/globals/models.py [hunk 8]
```diff
@@ -143,6 +143,8 @@ def __str__(self):
 class Faculty(models.Model):
     id = models.OneToOneField(ExtraInfo, on_delete=models.CASCADE, primary_key=True)
 
+    
+
```
Comment (adi0311) [inline FusionIIIT/applications/globals/models.py:147]: Don't add extra lines for no reason.

**Review by adi0311 — CHANGES_REQUESTED** [review]

File: FusionIIIT/applications/hr2/models.py [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+
+
+
+# table for employee details
+# Employee model
+class Employee(models.Model):
+
+    extra_info = models.OneToOneField(ExtraInfo, on_delete=models.CASCADE)
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
```
Comment (adi0311) [inline FusionIIIT/applications/hr2/models.py]: User is already present in ExtraInfo. Remove it as it is redundant.
Docstrings still not added. Add docstring to every models and views.

File: FusionIIIT/applications/hr2/views.py [hunk 10]
```diff
@@ -0,0 +1,10 @@
+from django.shortcuts import render
+
+# Create your views here.
+
+def hr2_index(request):
+    template = 'hr2Module/hr2_index.html'
+    return render(request,template)
+def hrAdmin(request):
```
Comment (adi0311) [inline FusionIIIT/applications/hr2/views.py]: You are instructed to read PEP-8 and follow those guidelines.

**Comment by Palash1999** [comment]
This PR is not maintained anymore, please refer to this new pull request of HR-2 Module [here](https://github.com/FusionIIIT/Fusion/pull/357).

**Pull request closed without merging**